### PR TITLE
PHP Update -- Batch support, imagefeatures, private cloud, and auth

### PR DESCRIPTION
**Pull request by madisonmay**
A rather large update to our PHP library, slowly moving towards something a bit more fully featured.  

Still to do:

- [x] Documentation updates
- [x] Batch API tests
- [x] Private cloud configuration tests
- [x] Skip some tests if batch API credentials aren't provided

### CODE REVIEW

**Comment by madisonmay** [comment]
Status: code review --> rebase --> merge.

File: tests/IndicoIoTest.php [hunk 1]
```diff
@@ -17,112 +32,163 @@ public function testPoliticalWhenGivenTheRightParameters()
...
+    {
+        $data = IndicoIo::text_tags('On Monday, the president will be ...');
+        $keys_result = array_keys($data);
+        $this->assertEquals(count($keys_result), 111);
+    }
 
-        $data = \IndicoIo\IndicoIo::language('bonsoir les jeunes !');
+    public function testFerWhenGivenTheRightParameters()
+    {
+        $humour_expected = array('Angry', 'Sad', 'Neutral', 'Surprise', 'Fear', 'Happy');
```
Comment (anneCarlson) [inline tests/IndicoIoTest.php]: This is a weird variable name here.

**Comment by anneCarlson** [comment]
Looks good to me! Merge at will.

**Comment by anneCarlson** [comment]
I didn't see versioning updates, actually. Would be good to add.

**Comment by madisonmay** [comment]
Oddly enough PHP's versioning is done via an online GUI on upload.

**Pull request merged**